### PR TITLE
update documentation

### LINKED ISSUE #292: Update models.py documentation
A lot of documentation in `models.py` is outdated, particularly in the `ReplayStage` stage. These should be updated.

### PR DESCRIPTION
**Pull request by raviddog**
fixes #292 

if the parser comments are too much i can cut back on them, but I really like the idea of having the information right next to the code, even if it is duplicated

### CODE REVIEW

File: project/thscoreboard/replays/models.py
Comment (jgdhs27) [inline project/thscoreboard/replays/models.py:1]: The docstring of `def CleanUp(cls, now: datetime.datetime) -> None:` says that an int is returned, this needs to be updated too

File: project/thscoreboard/replays/replay_parsing.py [hunk 1]
```diff
@@ -530,6 +544,7 @@ def _Parse12(rep_raw):
         if next_stage is not None:
             s.score = next_stage.score * 10
             s.power = next_stage.power
+            # piv is stored with extra precision, we trunctate the value to what is shown ingame
             s.piv = (math.trunc(next_stage.piv / 1000)) * 10
```
Comment (jgdhs27) [inline project/thscoreboard/replays/replay_parsing.py:548]: imo this should be extracted to a function `def convert_piv_stored_in_replay_to_displayed_piv(game_id, PIV: int) -> int`

File: project/thscoreboard/replays/replay_parsing.py [hunk 2]
```diff
@@ -112,11 +113,14 @@ def _Parse06(rep_raw):
 
     rep_stages = []
 
+    # TH06 stores stage data values from the start of the stage but score from the end
```
Comment (jgdhs27) [inline project/thscoreboard/replays/replay_parsing.py:116]: This comment is too far from the code it's describing

File: project/thscoreboard/replays/replay_parsing.py (hunk 1 above)
Comment (raviddog) [reply project/thscoreboard/replays/replay_parsing.py:548]: this is looking awfully similar to the game_fields format functions, should we be storing the PIV the same as in the file, and just converting it visually along with the rest of the data there?

File: project/thscoreboard/replays/replay_parsing.py [hunk 3]
```diff
@@ -427,6 +436,7 @@ def _Parse10(rep_raw):
 
     rep_stages = []
     
+    # TH10 stores all stage data values from the start of the stage so we need to retrieve them from the next stage
```
[the remark this answers is not in the document]
Comment (raviddog) [reply project/thscoreboard/replays/replay_parsing.py:439]: good idea

File: project/thscoreboard/replays/replay_parsing.py [hunk 4]
```diff
@@ -901,13 +933,16 @@ def _Parse18(rep_raw) -> ReplayInfo:
 
     rep_stages = []
     
+    # TH18 contains both stage start and end data for each stage
+    # However the end-of-stage data is inaccurate as explained below, so the next stage data is still required
```
[the remark this answers is not in the document]
Comment (raviddog) [reply project/thscoreboard/replays/replay_parsing.py:937]: My thought process was the code these comments are on are also linked anyway, but you're right this isn't needed

File: project/thscoreboard/replays/models.py [hunk 5]
```diff
@@ -329,9 +329,8 @@ def SetForeignKeysFromConstantModels(self, c: ReplayConstantModels):
 class ReplayStage(models.Model):
     """ Represents the end-of-stage data for a stage split for a given replay
         The data may not directly correspond to how it is stored in-game, since some games store it differently
-        Most games store the values from the start of the stage
-        TH07 stores the values from the end of the stage
-        TH08 is weird in that the score is stored from end of stage, but everything else is from the start
+        All differences/quirks in how this data is stored in-game is handled in the relevant replay_parsing.Parse() functions
```
[the remark this answers is not in the document]
Comment (raviddog) [reply project/thscoreboard/replays/models.py:332]: This goes along with the PIV function comment, currently there's 2 places where data is getting transformed from the replay file, and its not consistent whether fields in the DB are pre- or post transform. Documenting them in the models would eventually be forgotten as you say, but we need some form of either consistency in the data, or somewhere to say what the ruling on how data gets entered into each field is. I'm not sure which path we should take.

**Comment by raviddog** [comment]
I know we don't agree on a lot of this but I do think this is an improvement over what's currently in the codebase.  You can reopen the issue if you feel further changes are warranted but lets get this moving forward a bit.

**Review by jgdhs27 — APPROVED** [review]

**Pull request merged**